### ISSUE TITLE
Attempting backup results in unknown user admin

### ISSUE THREAD

**Issue by wolcen** [YunoHost/issues#1695]
Hello!

When I attempt to create a backup I receive the "Unknown user admin" from the web UI. This is not impacted by the selections made.

The admin user password can be changed without complaint with `yunohost tools adminpw`.

I'm still trying to understand how the pieces of this system fit together....hopefully I'll be able to help :shrug:

In the meantime, it doesn't look like anything interesting is in either log file.

cli:
```
2020-12-16 07:21:41,381 DEBUG    moulinette.core acquire - acquiring lock...
2020-12-16 07:21:41,391 DEBUG    moulinette.core acquire - lock has been acquired
2020-12-16 07:21:41,409 DEBUG    moulinette.actionsmap process - loading python module yunohost.backup took 0.017s
2020-12-16 07:21:41,409 INFO     moulinette.actionsmap process - processing action [29614.1]: yunohost.backup.create with args={'description': None, 'apps': None, 'system': None, 'output_directory': None, 'methods': None, 'no_compress': False, 'name': None}
2020-12-16 07:21:41,411 DEBUG    yunohost.hook hook_list - [29614.1] No custom hook for action 'backup' in /etc/yunohost/hooks.d/
2020-12-16 07:21:41,411 DEBUG    yunohost.backup _collect_system_files - [29614.1] Running backup hooks...
2020-12-16 07:21:41,412 DEBUG    yunohost.hook hook_list - [29614.1] No custom hook for action 'backup' in /etc/yunohost/hooks.d/
2020-12-16 07:21:41,414 DEBUG    yunohost.hook _hook_exec_bash - [29614.1] Executing command 'sh -c YNH_CWD=/home/yunohost.backup/tmp/20201216-122141 YNH_STDINFO=/tmp/tmp46Dl89/stdinfo YNH_BACKUP_CSV=/tmp/backupcsv_KXwwxx YNH_BACKUP_DIR=/home/yunohost.backup/tmp/20201216-122141 YNH_STDRETURN=/tmp/tmpIckLNh/stdreturn YNH_INTERFACE=cli BASH_XTRACEFD=7 /bin/bash -x "/usr/share/yunohost/hooks/backup/05-conf_ldap" /home/yunohost.backup/tmp/20201216-122141 7>&1'
etc...
```

via web:
```
2020-12-16 07:36:17,270 DEBUG    geventwebsocket.handler run_application - Initializing WebSocket
2020-12-16 07:36:17,270 DEBUG    geventwebsocket.handler upgrade_websocket - Validating WebSocket request
2020-12-16 07:36:17,270 DEBUG    geventwebsocket.handler upgrade_connection - Attempting to upgrade connection
2020-12-16 07:36:17,271 DEBUG    geventwebsocket.handler upgrade_connection - WebSocket request accepted, switching protocols
2020-12-16 07:36:17,277 DEBUG    geventwebsocket.handler run_application - Initializing WebSocket
2020-12-16 07:36:17,278 DEBUG    geventwebsocket.handler upgrade_websocket - Validating WebSocket request
2020-12-16 07:36:17,278 DEBUG    geventwebsocket.handler upgrade_websocket - Can only upgrade connection if using GET method.
2020-12-16 07:36:17,278 DEBUG    moulinette.authenticator.ldap __init__ - initialize authenticator 'default' with: uri='ldap://localhost:389', base_dn='dc=yunohost,dc=org', user_rdn='cn=admin,dc=yunohost,dc=org'
2020-12-16 07:36:17,279 DEBUG    moulinette.core acquire - acquiring lock...
2020-12-16 07:36:17,279 DEBUG    moulinette.core acquire - lock has been acquired
2020-12-16 07:36:17,279 DEBUG    moulinette.actionsmap process - loading python module yunohost.backup took 0.000s
2020-12-16 07:36:17,279 INFO     moulinette.actionsmap process - processing action [449.230]: yunohost.backup.create with args={'methods': None, 'name': None, 'description': None, 'apps': None, 'system': None, 'output_directory': None, 'no_compress': False}
2020-12-16 07:36:17,280 DEBUG    moulinette.actionsmap process - action [449.230] executed in 0.001s
2020-12-16 07:36:17,280 DEBUG    moulinette.core release - lock has been released
2020-12-16 07:36:17,281 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2020-12-16 07:36:17] "POST /backup HTTP/1.1" 400 145 0.003313
2020-12-16 07:36:17,281 DEBUG    geventwebsocket.handler close - Closed WebSocket
2020-12-16 07:36:17,281 DEBUG    geventwebsocket.handler close - Failed to write closing frame -> closing socket
2020-12-16 07:36:17,281 DEBUG    geventwebsocket.handler close - Closed WebSocket```
```

Thankfully, running `yunohost backup create` works fine (and the backups appear in UI), so clearly something up with the handling/calling of the moulinette api.

**Comment by zamentur** [comment]
Have you installed the borg_ynh app ?

What's your yunohost version ? `yunohost --version`

**Comment by wolcen** [comment]
I have not yet tried borg nor any of backup-related apps.

```
yunohost:
  repo: stable
  version: 4.0.8.3
yunohost-admin:
  repo: stable
  version: 4.0.4
moulinette:
  repo: stable
  version: 4.0.3
ssowat:
  repo: stable
  version: 4.0.5
```

**Comment by zamentur** [comment]
> When I attempt to create a backup I receive the "Unknown user admin" from the web UI.

Could we have screenshot or full log of this ?

What's the result of 
```
id admin
systemctl status nscd
systemctl status nslcd

```

Have you some warning inside your diagnostics ?

**Comment by wolcen** [comment]
I've noticed that passwords can show up in the log, so I reviewed the applicable section from what api/cli logs I could identify. What other log content is requested? FWIW, the server returns a 400 and the response content is just the literal string shown as the error.

In the system review, yes, there are some warnings/errors, but only IP6/DNS/email or internet-accessibility-related, which are sensible as I've intentionally not set those items up (thus far).

- Internet Connectivity:  The server does not have working IPv6. 
- DNS: no wildcard, CAA, email, xmpp [HTTPS is working fine w/LetsEncrypt]
- Ports (all blocked except http(s))
- Email issues should be obvious.

![image](https://user-images.githubusercontent.com/7354450/102403130-ce6edd00-3fb3-11eb-97c3-623584f8bb90.png)

![image](https://user-images.githubusercontent.com/7354450/102404233-64573780-3fb5-11eb-91c8-7948bef6b8b8.png)

I'm away from the system right now and wetty is having an issue as well [I'll document that separately when I can after further investigation]. As I'd not opened up ssh quite yet, I'll probably have a limited ability to investigate for a couple days. This is not urgent, from my perspective, but I appreciate the quick responses!

One other thing I noticed is that an expired cookie is being returned (and rejected by the browser) from every nginx response: SSOwAuthRedirect=; Path=/yunohost/sso/; Expires=Thu, 01 Jan 1970 00:00:00 UTC; Secure; HttpOnly; SameSite=Lax

PS - I'll be doing a demo of your wonderful project at [Agaric coop's Show & Tell](https://agaric.coop/show-and-tell) tomorrow afternoon. As I'm away from the system and going to do this demo, I'm hesitant to reboot until afterwards, but am planning to - I don't think it's been restarted at all yet, and I'm curious if that'll have an effect.

**Comment by zamentur** [comment]
What's the result of
```
id admin
slapcat | grep "uid: admin"
grep admin /etc/shadow # Feel free to modify the hash or password if one appears
```

**Comment by wolcen** [comment]
TL/DR: Restarted yunohost-api and that resolved the problem; Next attempt at a backup worked fine.

-----
I can't try the other commands right now, but if it matters, I had done `getent passwd admin` and received an expected result before leaving home. The ldap account can be found, and ssh as that user was fine. I did NOT create a "normal" admin system user. Unfortunately, I hadn't looked at anything related to nscd/nslcd.

Given the suggestion to check those services, I was curios to see if they were listed on the services page. While I was on that page, not finding them, but noticing yunohost-api was available, I decided to try restarting it.

If you wanted to request anything else, certainly do and I can try to do more digging back when I return home (e.g. perhaps yunohost-api had reported some issue ealier on). Thanks!
